.github/workflows/check-learning-path-links.yml

```diff
@@ -0,0 +1,79 @@
+name: 'Check Learning Path Links'
+on:
+  schedule: # Run once a month
+    - cron: '0 0 1 * *'
+  workflow_dispatch:
+
+permissions: {}
+  
+jobs:
+  check-learning-path-links:
+    name: 'Check Learning Path Links'
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+
+    steps:
+      - name: Checkout head
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+          fetch-depth: 0 # Fetch the entire repo for the below git commit graph operations
+          ref: main
+
+      - name: Get current SHA
+        id: get_new_sha
+        run: |
+          new_sha=$(git rev-parse HEAD)
+          echo "new_sha=$new_sha" >> $GITHUB_ENV
+
+      - name: Get previous update SHA
+        id: get_sha
+        run: |
+          cd .github
+          prev_sha=$(cat learning-path-sha.txt)
+          echo "prev_sha=$prev_sha" >> $GITHUB_ENV
+          cd ../
+
+      - name: Checkout head
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: true
+          ref: main
+
+      - name: Checkout previous update
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+          ref:  ${{ env.prev_sha }}
+          path: prev
+
+      - name: Get changed files
+        run: |
+          changed_source_files=$(git diff-tree --no-commit-id --name-only -r "$prev_sha" "$GITHUB_SHA" | { grep "**.cs$" || test $? = 1; })
+          echo "Files to validate: '${changed_source_files}'"
+          echo "updated_files=$(echo ${changed_source_files})" >> $GITHUB_ENV
+
+      - name: Check Learning Path Links
+        id: check-links
+        uses: ./.github/actions/learning-path-staleness-check
+        with:
+          repoURLToSearch: 'https://github.com/dotnet/dotnet-monitor'
+          learningPathsDirectory: 'documentation/learningPath'
+          changedFilePaths: ${{ env.updated_files }}
+          sourceDirectoryName: 'src'
+          oldHash: ${{ env.prev_sha }}
+          newHash: ${{ env.new_sha }}
+          learningPathHashFile: '.github/learning-path-sha.txt'
+
+      - name: Open PR
+        uses: ./.github/actions/open-pr
+        with:
+          files_to_commit: ${{ steps.check-links.outputs.modifiedFilesToCommit }}
+          title: "[REQUIRES MANUAL REVIEW] Update Learning Paths"
+          commit_message: Update Learning Paths
+          body: This PR was auto generated and will not be automatically merged in - adjustments should be made manually as-needed.<br />${{ steps.check-links.outputs.outputText }}
+          branch_name: learningPathUpdates/${{ env.new_sha }}
+          fail_if_files_unchanged: true
```

Comment:
This doesn't need to be addressed in this PR, but it should be addressed soon.

This will eventually lead to workflow failures if there were no new commits to a repro between workflow runs (so `HEAD` and `learning-path-sha` are the same). I'd suggest breaking up this pipeline into distinct jobs (one to calculate the set of files changed between commits, and one to update the links / open a pr).

Reply:
I was aware of this, but wasn't too worried given 1) that would suggest no new commits over the course of a month (which is incredibly unlikely at least for our repo), and 2) the workflow failing wouldn't have any consequences. But yes, this could be addressed later.